test/jdk/java/net/httpclient/http2/UserInfoTest.java

```diff
@@ -0,0 +1,99 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.net.SimpleSSLContext;
+import org.testng.annotations.Test;
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+
+/**
+ * @test
+ * @bug 8292876
+ * @library /test/lib server
+ * @modules java.base/sun.net.www.http
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ * @run testng UserInfoTest
+ */
+
+public class UserInfoTest {
+
+    static class Http2TestHandler implements Http2Handler{
+        @Override
+        public void handle(Http2TestExchange e) throws IOException {
+            InputStream is = e.getRequestBody();
+            StringBuilder sb = new StringBuilder();
+            for (int ch; (ch = is.read()) != -1; ) {
+                sb.append((char) ch);
+            }
+            is.readAllBytes();
+            is.close();
+            if(e.getRequestURI().getAuthority().contains("user@")){
+                e.sendResponseHeaders(500, -1);
+            } else {
+                e.sendResponseHeaders(200, -1);
+            }
+        }
+    }
+
+    private static Http2TestServer createServer(SSLContext sslContext) throws Exception {
+        Http2TestServer http2TestServer = new Http2TestServer("localhost",true, sslContext);
+        Http2TestHandler handler = new Http2TestHandler();
+        http2TestServer.addHandler(handler, "/");
+        return http2TestServer;
+    }
+
+    @Test
+    public void main() throws Exception {
+        SSLContext sslContext = new SimpleSSLContext().get();
+        Http2TestServer server = createServer(sslContext);
+        int port = server.getAddress().getPort();
+
+        server.start();
+
+        HttpClient client = HttpClient
+                .newBuilder()
+                .sslContext(sslContext)
+                .build();
+
+        URI uri = new URI("https://user@127.0.0.1:" + Integer.toString(port));
+
+        HttpRequest request = HttpRequest.newBuilder(uri)
+                .GET()
+                .build();
+
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        if(response.statusCode()==500){
+            throw new RuntimeException("Test Failed : " + response.uri().getAuthority());
+        }
+        server.stop();
```

Comment:
Since the test is running in agentvm mode then it is better to stop the test server once the test method is completed execution: One possibility is to wrap http request code and the status code check into `try {} finally{}` and stop the server even if the test failed with `RuntimeException`.